### PR TITLE
feat: add upgradeability

### PR DESCRIPTION
**Pull request by byteZorvin**
- issue #35 

This pr add upgradeability using the Openzeppelin's upgradeable component

### CODE REVIEW

File: src/appchain.cairo [hunk 1]
```diff
@@ -197,4 +206,15 @@ mod appchain {
                 );
         }
     }
+
+    #[abi(embed_v0)]
+    impl UpgradeableImpl of IUpgradeable<ContractState> {
+        fn upgrade(ref self: ContractState, new_class_hash: ClassHash) {
+            // This function can only be called by the owner
```
Comment (glihm) [inline src/appchain.cairo:213]: Comment that could be removed as code is self-explanatory.

File: src/appchain.cairo [hunk 2]
```diff
@@ -197,4 +206,15 @@ mod appchain {
...
+            self.ownable.assert_only_owner();
+
+            // Replace the class hash upgrading the contract
```
Comment (glihm) [inline src/appchain.cairo:216]: Same here, comment could be removed as very self-explanatory.

**Review by glihm — CHANGES_REQUESTED** [review]

File: src/appchain.cairo (hunk 1 above)
Comment (byteZorvin) [reply src/appchain.cairo:213]: updated

File: src/appchain.cairo (hunk 2 above)
Comment (byteZorvin) [reply src/appchain.cairo:216]: updated

**Review by glihm — APPROVED** [review]
We don't have a test for that, but could be handled in an other PR if you want.
Thanks for the contribution @byteZorvin! 🚀

**Pull request merged**